packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx

```diff
@@ -118,3 +126,58 @@ export const useLocalVideoStartTrigger = (isLocalVideoAvailable: boolean, should
     }
   }, [shouldTransition, isButtonStatusSynced, isPreviewCameraOn, isLocalVideoAvailable, mediaGalleryHandlers]);
 };
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ */
+const useParticipantChangedAnnouncement = (): string => {
+  const adapter = useAdapter();
+  const locale = useLocale().strings.call;
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  /**
+   * state to track whether there is currently a timer set in the MediaGallery
+   */
+  const [timeoutState, setTimeoutState] = useState<ReturnType<typeof setTimeout>>();
+
+  const setParticipantEventString = (string: string): void => {
+    setAnnouncerString('');
+
+    if (timeoutState) {
+      clearTimeout(timeoutState);
+      setTimeoutState(undefined);
+    }
+    setTimeoutState(
+      /**
+       * These set timeouts are needed to clear the announcer string in case we have multiple
+       * participants join. Since the narrator will only announce the string in the
+       * Announcer component should the string change.
+       */
+      setTimeout(() => {
+        setAnnouncerString(string);
+        setTimeoutState(undefined);
+      }, 500)
```

Comment:
no arbitrary constants nit, pull this out as a constant and give a name and a comment for its purpose (you can probably move the comment from the setTimeout as it becomes redundant if the constant is named and commented), e.g.
```tsx
// Delay before telling a screen reader to announce a participant joined/left event.
// The delay is needed ...
const PARTICIPANT_ANNOUNCEMENT_DELAY = 500
```